src/Raven.Server/Routing/RequestRouter.cs

```diff
@@ -223,18 +223,28 @@ public static bool TryGetClientVersion(HttpContext context, out Version version)
         public static void AssertClientVersion(HttpContext context, Exception innerException)
         {
             // client in this context could be also a follower sending a command to his leader.
-            if (TryGetClientVersion(context, out var clientVersion))
-            {
-                var currentServerVersion = RavenVersionAttribute.Instance;
+            if (TryGetClientVersion(context, out var clientVersion) == false) 
+                return;
 
-                if (currentServerVersion.MajorVersion != clientVersion.Major || currentServerVersion.BuildVersion < clientVersion.Revision || currentServerVersion.BuildVersion == ServerVersion.DevBuildNumber || (clientVersion.Revision >= 40 && clientVersion.Revision < 50))
-                {
-                    throw new ClientVersionMismatchException(
-                        $"Failed to make a request from a newer client with build version {clientVersion} to an older server with build version {RavenVersionAttribute.Instance.AssemblyVersion}.{Environment.NewLine}" +
-                        $"Upgrading this node might fix this issue.",
-                        innerException);
-                }
-            }
+            if(CheckClientVersionAndWrapException(clientVersion, ref innerException) == false)
+                throw innerException;
+        }
+
+        public static bool CheckClientVersionAndWrapException(Version clientVersion, ref Exception innerException)
+        {
+            var currentServerVersion = RavenVersionAttribute.Instance;
+
+            if (currentServerVersion.MajorVersion == clientVersion.Major &&
+                currentServerVersion.BuildVersion >= clientVersion.Revision &&
+                currentServerVersion.BuildVersion != ServerVersion.DevBuildNumber &&
+                (clientVersion.Revision < 40 || clientVersion.Revision >= 50)) 
```

Comment:
Why this line?

Reply:
If I understand it right it is to check if the client unofficially built (in a development environment)

Reply:
You are excluding only 40 .. 50 here?

Reply:
This is the range of v4.x other versions will be check by the first condition

Reply:
Problem is that the 6.0 client will be accepted here, accidently.

Reply:
How? The major is not equal?

Reply:
I don't like that line. It make no sense to me. 
The dev version we use is either 40 or 50 for dev, no? In that case, check for those using equality, please.